catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -13,72 +13,104 @@
       return convert(stream);
     } catch (IOException e) {
       LOGGER.debug("IO exception during conversion of {}", xml, e);
       throw new ValidationExceptionImpl(
-          e, Collections.singletonList("IO exception during conversion"), new ArrayList<String>());
+          e, Collections.singletonList("IO exception during conversion"), new ArrayList<>());
+    }
+  }
+
+  public String convert(InputStream xml) throws ValidationException {
+    Object parsedObject = parseXml(xml);
+
+    if (parsedObject instanceof Envelope) {
+      parsedObject = JTS.toGeometry((Envelope) parsedObject);
     }
+
+    if (parsedObject instanceof Geometry) {
+      try {
+        Geometry geometry = convertCRS((Geometry) parsedObject);
+        return WKT_WRITER.get().write(geometry);
+      } catch (TransformException e) {
+        LOGGER.debug("Failed to transform geometry to lon/lat", e);
+        throw new ValidationExceptionImpl(
+            e,
+            Collections.singletonList("Cannot transform geometry to lon/lat"),
+            new ArrayList<>());
+      }
+    }
+    LOGGER.debug("Unknown object parsed from GML and unable to convert to WKT");
+    throw new ValidationExceptionImpl(
+        "", Collections.singletonList("Couldn't not convert GML to WKT"), new ArrayList<>());
   }
 
-  @SuppressWarnings("unchecked")
-  public String convert(InputStream xml) throws ValidationExceptionImpl {
-    AbstractGeometryType geometry = null;
+  public Object parseXml(InputStream xml) throws ValidationException {
     try {
-      JAXBElement<AbstractGeometryType> jaxbElement =
-          parser.unmarshal(configurator, JAXBElement.class, xml);
-      geometry = jaxbElement.getValue();
-      GML311ToJTSGeometryConverter geometryConverter = new GML311ToJTSGeometryConverter();
-      Geometry jtsGeo =
-          geometryConverter.createGeometry(new DefaultRootObjectLocator(jaxbElement), geometry);
-      WKTWriter wktWriter = new WKTWriter();
-      return wktWriter.write(jtsGeo);
-    } catch (ParserException e) {
-      LOGGER.debug("Cannot parse gml", e);
+      return parser.parse(xml);
+    } catch (ParserConfigurationException | IOException e) {
+      LOGGER.debug("Failed to read gml InputStream", e);
+      throw new ValidationExceptionImpl(
+          e, Collections.singletonList("Cannot read gml"), new ArrayList<>());
+    } catch (SAXException e) {
+      LOGGER.debug("Failed to parse gml xml", e);
       throw new ValidationExceptionImpl(
-          e, Collections.singletonList("Cannot parse gml"), new ArrayList<String>());
-    } catch (ConversionFailedException e) {
-      LOGGER.debug("Cannot convert gml311 geo object {} to jts", geometry, e);
+          e, Collections.singletonList("Cannot parse gml xml"), new ArrayList<>());
+    }
+  }
+
+  private synchronized Geometry convertCRS(Geometry geometry)
+      throws ValidationException, TransformException {
+    return JTS.transform(geometry, getLatLonTransform());
```

Comment:
❓ It looks like both JTS and CRS classes either create local objects to use or are already synchronized where need be -- was there another reason we need to have this method synchronized?

Reply:
It was originally for the `MathTransform` class, but i don't think it needs to be synchronized anymore since we create it each time we need it now.